src/plugins/utils/http-utils.ts

```diff
@@ -123,15 +122,20 @@ export const createXRaySubsegment = (
 };
 
 export const requestInfoToHostname = (request: Request | URL | string) => {
-    let hostname = '';
-    if ((request as URL).hostname) {
-        hostname = (request as URL).hostname;
-    } else if ((request as Request).url) {
-        hostname = getHost((request as Request).url);
-    } else {
-        hostname = getHost(request.toString());
+    try {
+        if ((request as URL).hostname) {
+            return (request as URL).hostname;
+        } else if ((request as Request).url) {
+            return new URL((request as Request).url).hostname;
+        } else {
+            return new URL(request.toString()).hostname;
```

Comment:
why the need for the `.toString`?

Reply:
TypeScript likes the type to be explicit.

Reply:
can do:

```ts
             return new URL(<String>request).hostname;
```

but I guess that is a bit hacky :-|